File: api/src/main/java/clases/Coleccion.java
package clases;

import java.util.ArrayList;

public class Coleccion {

    public Coleccion(int idUsuario, String titulo, String imagenBase64) {

        setIdUsuario(idUsuario);
        setTitulo(titulo);
        setImagenBase64(imagenBase64);
    }

    public Coleccion() {
        this(0, "", null);
    }

    private int id;

    public void setId(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    private int idUsuario;

    public void setIdUsuario(int idUsuario) {
        this.idUsuario = idUsuario;
    }

    public int getIdUsuario() {
        return idUsuario;
    }

    private String titulo;

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public String getTitulo() {
        return titulo;
    }

    private String imagenBase64;

    public void setImagenBase64(String imagenBase64) {
        this.imagenBase64 = imagenBase64;
    }

    public String getImagenBase64() {
        return imagenBase64;
    }

    @Override
    public String toString() {
        return "Coleccion{id=" + id +
                ", titulo=" + titulo +
                ", idUsuario=" + idUsuario + "}";
    }
}

File: wikz_web/modelos/coleccion.php
<?php

class Coleccion
{
    private int $id;
    private int $idUsuario;
    private string $titulo;
    private array $elementos; // Array de Publicacion
    private ?string $imagenBase64;

    // Constructor principal
    public function __construct(
        int $idUsuario = 0,
        string $titulo = "",
        array $elementos = [],
        ?string $imagenBase64 = null
    ) {
        $this->idUsuario = $idUsuario;
        $this->titulo = $titulo;
        $this->elementos = $elementos;
        $this->imagenBase64 = $imagenBase64;
    }

    // ===== GETTERS & SETTERS =====

    public function getId(): int
    {
        return $this->id;
    }

    public function setId(int $id): void
    {
        $this->id = $id;
    }

    public function getIdUsuario(): int
    {
        return $this->idUsuario;
    }

    public function setIdUsuario(int $idUsuario): void
    {
        $this->idUsuario = $idUsuario;
    }

    public function getTitulo(): string
    {
        return $this->titulo;
    }

    public function setTitulo(string $titulo): void
    {
        $this->titulo = $titulo;
    }

    /**
     * @return Publicacion[]
     */
    public function getElementos(): array
    {
        return $this->elementos;
    }

    /**
     * @param Publicacion[] $elementos
     */
    public function setElementos(array $elementos): void
    {
        $this->elementos = $elementos;
    }

    public function getImagenBase64(): ?string
    {
        return $this->imagenBase64;
    }

    public function setImagenBase64(?string $imagenBase64): void
    {
        $this->imagenBase64 = $imagenBase64;
    }

    // ===== toString =====
    public function __toString(): string
    {
        return "Coleccion{id={$this->id}, titulo={$this->titulo}, elementos=" .
            count($this->elementos) .
            ", idUsuario={$this->idUsuario}}";
    }
}
